modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -227,6 +226,13 @@ private void populateTarget(Target target, OMElement sourceEle) {
                 } catch (JaxenException e) {
                     handleException("Invalid XPath expression: " + xpathAttr);
                 }
+                if (target.getAction().equals(Target.ACTION_REPLACE) && (target.getXpath() instanceof SynapseJsonPath)
+                        && ("$".equals(((SynapseJsonPath) target.getXpath()).expression) ||
```

Comment:
Shall we extract target.getXpath as targetXPath and target.getXpath()).expression as targetExpression for better clarity?

Reply:
Fixed in https://github.com/wso2/wso2-synapse/pull/1445